evalml/tests/automl_tests/test_automl.py

```diff
@@ -5319,3 +5320,126 @@ def test_init_create_holdout_set(caplog):
             problem_type="binary",
             holdout_set_size=-0.1,
         )
+
+
+def test_ordinal_encoder_in_automl(X_y_ordinal_regression):
+    X, y = X_y_ordinal_regression
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="regression",
+        objective="R2",
+        random_seed=0,
+        n_jobs=1,
+    )
+    automl.search()
+
+    num_features_in_X_t = 0
+    for i in range(1, len(automl.rankings)):
+        pipeline = automl.get_pipeline(i)
+        if not pipeline._is_fitted:
+            pipeline.fit(X, y)
+        X_t = pipeline.transform_all_but_final(X)
+
+        assert "Ordinal Encoder" in pipeline.name
+        if "day" in pipeline._get_feature_provenance():
+            assert "day_ordinal_encoding" in X_t.columns
+            num_features_in_X_t += 1
+    assert num_features_in_X_t
+
+
+def test_ordinal_encoder_not_used_in_automl_if_no_ordinal_columns_present(
+    X_y_categorical_regression,
+):
+    X, y = X_y_categorical_regression
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="regression",
+        objective="R2",
+        random_seed=0,
+        n_jobs=1,
+    )
+    automl.search()
+
+    for i in range(1, len(automl.rankings)):
+        pipeline = automl.get_pipeline(i)
+        if not pipeline._is_fitted:
+            pipeline.fit(X, y)
+        X_t = pipeline.transform_all_but_final(X)
+
+        assert "Ordinal Encoder" not in pipeline.name
+        assert "day_ordinal_encoding" not in X_t.columns
+
+
+def test_add_ordinal_encoder_to_pipeline(X_y_ordinal_regression):
+    X, y = X_y_ordinal_regression
+
+    pipeline = RegressionPipeline(
+        component_graph=[
+            "Imputer",
+            "Ordinal Encoder",
+            "One Hot Encoder",
+            "DateTime Featurizer",
+            "Standard Scaler",
+            "Linear Regressor",
+        ],
+    )
+    pipeline.fit(X, y)
+    X_t = pipeline.transform_all_but_final(X)
+    # Confirm Ordinal Encoder is in the component graph
+    assert "Ordinal Encoder" in pipeline.component_graph.component_dict.keys()
+    # Confirm the ordinal feature actually gets encoded
+    assert X_t.ww.columns["day_ordinal_encoding"].is_numeric
+    assert "day" not in X_t.columns
+
+
+def test_ordinal_encoder_not_used_if_ohe_first(X_y_ordinal_regression):
```

Comment:
I think I added this initially to confirm that we needed the ordinal encoder to come first because of this behavior. I see how it probably isn't necessary other than as a form of documenting behavior/explaining why the ordinal encoder has to come first, so I'm happy to remove!